tests/testCentroidChecker.py

```diff
@@ -0,0 +1,171 @@
+        config.plugins = [self.algName]
+        config.slots.apFlux = None
+        config.slots.calibFlux = None
+        config.slots.instFlux = None
+        config.slots.modelFlux = None
+        config.slots.psfFlux = None
+        config.slots.shape = None
+        self.config = config
+
+    def tearDown(self):
+        del self.config
+        del self.dataset
+
+    def testNoError(self):
+        schema = self.dataset.makeMinimalSchema()
+        self.config.slots.centroid = "truth"
+        task = lsst.meas.base.SingleFrameMeasurementTask(schema=schema, config=self.config)
+        exposure, cat = self.dataset.realize(noise=100.0, schema=schema)
+        task.run(cat, exposure)
+        source = cat[0]
+        self.assertFalse(source.get("test_Centroider_flag"))
+        self.assertFalse(source.get("test_Centroider_flag_resetToPeak"))
+
+    #   test if a slight move of the centroid (but inside the footprint) will trigger
+    #   the distance error
+    def testCentroidDistance(self):
+        schema = self.dataset.makeMinimalSchema()
+        self.config.slots.centroid = "truth"
+        self.config.plugins[self.algName].moveX = -2
+        self.config.plugins[self.algName].dist = 1
+        task = lsst.meas.base.SingleFrameMeasurementTask(schema=schema, config=self.config)
+        exposure, cat = self.dataset.realize(noise=100.0, schema=schema)
+        source = cat[0]
+        task.run(cat, exposure)
+        self.assertTrue(source.get("test_Centroider_flag"))
+        self.assertTrue(source.get("test_Centroider_flag_resetToPeak"))
+        self.assertEqual(source.getFootprint().getPeaks()[0].getFx(), source.get("test_Centroider_x"))
+
+    #   test if moving the centroid completely outside of the footprint will trigger
+    #   the move back to the first peak
+    def testCentroidOutsideFootprint(self):
+        schema = self.dataset.makeMinimalSchema()
+        self.config.slots.centroid = "truth"
+        self.config.plugins[self.algName].moveX = -30
+        task = lsst.meas.base.SingleFrameMeasurementTask(schema=schema, config=self.config)
+        exposure, cat = self.dataset.realize(noise=100.0, schema=schema)
+        source = cat[0]
+        task.run(cat, exposure)
+        print source.get("test_Centroider_flag")
+        self.assertTrue(source.get("test_Centroider_flag"))
+        self.assertTrue(source.get("test_Centroider_flag_resetToPeak"))
+        self.assertEqual(source.getFootprint().getPeaks()[0].getFx(), source.get("test_Centroider_x"))
+
+
+def suite():
+    """Returns a suite containing all the test cases in this module."""
+
+    lsst.utils.tests.init()
+
+    suites = []
+    suites += unittest.makeSuite(FlagHandlerTestCase)
+    suites += unittest.makeSuite(lsst.utils.tests.MemoryTestCase)
+    return unittest.TestSuite(suites)
+
+
+def run(shouldExit=False):
+    """Run the tests"""
+    lsst.utils.tests.run(suite(), shouldExit)
+
+if __name__ == "__main__":
+    run(True)
```

Comment:
There's now a new recommended way to do the boilerplate at the end of Python tests.  You can find the instructions in the [SQR-012](https://sqr-012.lsst.io/) tech note.